ChiModules/LinearBoltzmannSolver/Tools/ksp_data_context.h

```diff
@@ -1,16 +1,38 @@
+#ifndef LBS_KSP_DATA_CONTEXT_H
+#define LBS_KSP_DATA_CONTEXT_H
 
+#include "../lbs_linear_boltzmann_solver.h"
+
+namespace LinearBoltzmann
+{
 
 //###################################################################
 /**This is a simple data structure of basically pointers to
  * objects needed in matrix free operations.*/
 struct KSPDataContext
 {
-  LinearBoltzmann::Solver* solver;
-  SweepChunk*      sweep_chunk;
-  int              group_set_num;
-  LBSGroupset*    groupset;
-  KSP              krylov_solver;
-  Vec              x_temp;
-  chi_mesh::sweep_management::SweepScheduler* sweepScheduler;
-  int last_iteration = -1;
-};
\ No newline at end of file
+  LinearBoltzmann::Solver& solver;
+  SweepChunk&              sweep_chunk;
+  LBSGroupset&             groupset;
+  Vec&                     operating_vector;
+  chi_mesh::sweep_management::SweepScheduler& sweepScheduler;
+  SourceFlags    lhs_scope;
+  size_t last_iteration = -1;
```

Comment:
Not sure if this is intentional or not. While wraparound is well defined by the standard, I argue that it's bad practice.  If it should accept signed values, use a signed type.

Reply:
Great catch Thx!